File: RailFenceTechnique/Encryption/RailFenceTechnique.scala
def RailFenceTechnique(plainText: String): String = {
    val formattedPlainText = plainText.replace(" ", "").toLowerCase
    val newWordEven = formattedPlainText.zipWithIndex.collect { case (c, i) if i % 2 == 0 => c }
    val newWordOdd = formattedPlainText.zipWithIndex.collect { case (c, i) if i % 2 == 1 => c }
    (newWordEven ++ newWordOdd).mkString
}

@main def main(): Unit = {
  print("Enter Plaintext: ")
  val plain_text = scala.io.StdIn.readLine()
  println(RailFenceTechnique(plain_text))
}

File: RailFenceTechnique/Encryption/RailFenceTechnique.js
function RailFenceTechnique(plainText) {
  const formattedPlainText = plainText.replace(/\s/g, "").toLowerCase();
  const newWordEven = [];
  const newWordOdd = [];
  for (let i = 0; i < formattedPlainText.length; i++) {
    if (i % 2 === 0) {
      newWordEven.push(formattedPlainText[i]);
    } else {
      newWordOdd.push(formattedPlainText[i]);
    }
  }
  const encryptedWord = newWordEven.concat(newWordOdd).join("");
  return "Encrypted Word: " + encryptedWord;
}

function read_input() {
  var plain_text = prompt("Enter the plain text: ").split("");
  console.log(RailFenceTechnique(plain_text.join("")));
}

read_input();
